cli/src/main.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate clap;
+#[macro_use]
+extern crate serde_derive;
+extern crate dirs;
+extern crate simple_logger;
+extern crate toml;
+#[macro_use]
+extern crate log;
+
+mod commands;
+mod utils;
+
+use clap::App;
+use commands::init;
+use log::LogLevel;
+
+fn main() {
+    let yaml = load_yaml!("cli.yaml");
```

Comment:
This assumes cli.yaml is in the current directory, which is not a valid assumption for a couple reasons: a) you might run the CLI from another directory; b) the CLI might be installed and the yaml file would not be installed in the bin directory (probably a share/lib directory depending on operating system).

It might be better to stick with the macro and/or code approach to avoid this complexity. (The macro approach is pretty good, IMO.)